docs/running/server/wes.rst

```diff
@@ -0,0 +1,272 @@
+            (default: 2).
+--opt ENGINE_OPTION
+            Specify the default parameters to be sent to the workflow engine for each run.  Accepts multiple values.
+
+            Example: ``toil server --opt=--logLevel=CRITICAL --opt=--workDir=/tmp``.
+
+.. _GA4GH docs on CORS: https://w3id.org/ga4gh/product-approval-support/cors
+
+
+Running on EC2
+--------------
+
+To run the server on a Toil leader instance on EC2:
+
+#. Launch a Toil cluster with the ``toil launch-cluster`` command with the AWS provisioner
+
+#. The Toil server requires an additional TCP port to open for HTTP(S) access. To add this inbound rule, open the AWS
+   console and go to the ``Security Groups`` section.  For example, on us-west2, this address would accessible at: ::
+
+    https://us-west-2.console.aws.amazon.com/ec2/v2/home?region=us-west-2#SecurityGroups
+
+#. Next, locate the security group created along with your cluster. It should be the same name as the cluster.
+
+#. Click on "Edit inbound rules". Then, add a rule to allow "0.0.0.0/0" (anyone) to access port 8080 (or the port where
+   the server will listen on), and click "Save rules".
+
+    .. image:: wes_ec2_1.png
+    .. image:: wes_ec2_2.png
+
+#. Now, you can SSH into your cluster and run the WES server with ``toil server`` on the Toil appliance.
+
+.. note::
+    To run the server in the background, you can run "``nohup toil server &``".
+
+
+.. _WESEndpointsOverview:
+
+WES API Endpoints
+-----------------
+
+As defined by the GA4GH WES API specification, the following endpoints with base path ``ga4gh/wes/v1/`` are supported
+by Toil:
+
++--------------------------------+--------------------------------------------------------+
+| GET /service-info              | Get information about the Workflow Execution Service.  |
++--------------------------------+--------------------------------------------------------+
+| GET /runs                      | List the workflow runs.                                |
++--------------------------------+--------------------------------------------------------+
+| POST /runs                     | Run a workflow. This endpoint creates a new workflow   |
+|                                | run and returns a ``run_id`` to monitor its progress.  |
++--------------------------------+--------------------------------------------------------+
+| GET /runs/{run_id}             | Get detailed info about a workflow run.                |
++--------------------------------+--------------------------------------------------------+
+| POST /runs/{run_id}/cancel     | Cancel a running workflow.                             |
++--------------------------------+--------------------------------------------------------+
+| GET /runs/{run_id}/status      | Get the status (overall state) of a workflow run.      |
++--------------------------------+--------------------------------------------------------+
+
+.. _WESSubmitWorkflow:
+
+Submitting a Workflow
```

Comment:
Adding some details on how to pass toil specific parameters for a CWL workflow would be helpful information. For example, how would I specify that I would like this run with mesos?

Reply:
I don't think there is a way to specify per-workflow Toil parameters; the WES spec doesn't seem to have anything like that. I suppose we could key on some Toil-defined WES "tags" if we wanted to be able to run workflows in different ways on the same server.

Presumably Mesos (or Kubernetes, if you set this up on a Toil-managed Kubernetes cluster) would be what you always use for all workflows submitted.

Reply:
Wait, I just didn't read. There's `workflow_engine_parameters` in which you can pass more parameters to Toil per-workflow. Though if the server is set to pass `--batchSystem=mesos` I don't think you could pass `--batchSystem=single_machine` and have it work.